src/inmanta/ast/statements/generator.py

```diff
@@ -252,7 +252,10 @@ def execute(self, requires: dict[object, object], resolver: Resolver, queue: Que
         var = self.base.execute(requires, resolver, queue)
 
         if not isinstance(var, (list, Unknown)):
-            raise TypingException(self, "A for loop can only be applied to lists and relations")
+            msg = "A for loop can only be applied to lists and relations."
+            if isinstance(var, NoneValue):
```

Comment:
I think it might be better to address this generically. i.e. no tailor-made message when it's `null`, just always tell the user what the value is.